### ISSUE TITLE
hjson-comments.extract/merge doesn't roundtrip array comments correctly.

### ISSUE THREAD

**Issue by ygraber** [hjson/hjson-js#40]
When trying to extract comments from the following hjson:
```
{
    a: 1,
    b: 2, # comment
    c: [
        'x',
        'y', #why?
        'z' // zee
    ]
}
```
you wind up with:
```
{
  a: 1
  b: 2 # comment
  c:
  [
    "x" #why?
    "y" // zee
    z
  ]
}
```
That is because extract is putting the comments into a sparse array `a` which only has elements for entries with comments (this will not hold up if the comments are themselves serialized to JSON). When merge is called, it is looping over all array elements and then transferring the comments to the current value.__COMMENTS__, but it is using push and losing all index information.

**Comment by ygraber** [comment]
proposed fix: https://github.com/hjson/hjson-js/pull/41

**Comment by dqsully** [comment]
Fixed by #41

**Comment by ygraber** [comment]
Any idea when you'll push this to  npmjs.com ?

**Comment by dqsully** [comment]
It should already be published under 3.1.2
